.github/workflows/atfe_nightly_build_and_test.yml

```diff
@@ -29,27 +30,54 @@ jobs:
       fail-fast: false  # Prevents one job failure from cancelling all
       matrix:
         include:
+          # build.sh
           - build_script: build.sh
             test_script: test.sh
+            target_os: Ubuntu-22.04-x86_64
             runner: ToolchainGroup_LargeRunner_Linux-x64
+
+          - build_script: build.sh
+            test_script: test.sh
+            target_os: Ubuntu-22.04-arm64
+            runner: ah-ubuntu_22_04-c7g_8x-100
+
+          - build_script: build.ps1
+            test_script: test.ps1
+            target_os: Windows-2022-x86_64
+            runner: ToolchainGroup_LargeRunner_Windows-x64
+
+          # build_llvmlibc_overlay.sh
           - build_script: build_llvmlibc_overlay.sh
             test_script: ''
+            target_os: Ubuntu-22.04-arm64
             runner: ah-ubuntu_22_04-c7g_8x-100
+
+          # build_newlib_nano_overlay.sh
           - build_script: build_newlib_nano_overlay.sh
             test_script: ''
+            target_os: Ubuntu-22.04-arm64
             runner: ah-ubuntu_22_04-c7g_8x-100
+
+          # build_newlib_overlay.sh
           - build_script: build_newlib_overlay.sh
             test_script: ''
+            target_os: Ubuntu-22.04-arm64
             runner: ah-ubuntu_22_04-c7g_8x-100
 
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Install dependencies
+      - name: Install Linux Ubuntu Dependencies
+        if: runner.os == 'Linux'
         run: bash ./arm-software/embedded/scripts/install_dependencies.sh
 
+      - name: Install Windows Dependencies
+        if: runner.os == 'Windows'
+        run: powershell ./arm-software/embedded/scripts/install_dependencies.ps1
```

Comment:
Would it make sense to have a `install_dependency_script` entry in the matrix strategy instead of having separate conditional steps? This way this script would be handled in a similar fashion to the build and test ones.